test/lib/bench_gateway/bench_gateway.go

```diff
@@ -282,6 +282,34 @@ func (gateway *BenchGateway) MakeRequest(
 	return client.Do(req)
 }
 
+// MakeRequestWithHeaderValues helper
+func (gateway *BenchGateway) MakeRequestWithHeaderValues(
+	method string, url string, headers zanzibar.Header, body io.Reader,
+) (*http.Response, error) {
+	client := gateway.httpClient
+
+	fullURL := "http://" + gateway.ActualGateway.RealHTTPAddr + url
+
+	req, err := http.NewRequest(method, fullURL, body)
+
+	// For each key, fetch every disparate header value and add
+	// it to the bench gateway request.
+	keys := headers.Keys()
+	for _, key := range keys {
+		if values, found := headers.Values(key); found {
+			for _, value := range values {
+				req.Header.Add(key, value)
+			}
+		}
+	}
+
+	if err != nil {
```

Comment:
any reason why this isn't right after where this happens?

Reply:
Just noticed its just how other ones are too but I don't think they need to be (or at least the new ones need to be).

Reply:
Good catch. I moved them to the correct location in both the old and new functions. Leaving it where it was also has the potential for a nil-pointer dereference with the following block accessing a property on `req`.